File: app/src/main/java/com/example/tradeup/data/repository/UserRepository.java
// File: src/main/java/com/example/tradeup/data/repository/UserRepository.java
package com.example.tradeup.data.repository;

import androidx.annotation.NonNull;
import com.example.tradeup.data.model.User;
import java.util.Map;
import java.util.concurrent.CompletableFuture;

public interface UserRepository {
    CompletableFuture<Void> createUserProfile(@NonNull User user);
    CompletableFuture<User> getUserProfile(String uid);
    CompletableFuture<Void> updateUserProfile(String uid, Map<String, Object> updates);
    CompletableFuture<Void> deactivateUser(String uid);
    CompletableFuture<Void> deleteUser(String uid);
    CompletableFuture<Void> blockUser(@NonNull String currentUserId, @NonNull String userToBlockId);
}

File: app/src/main/java/com/example/tradeup/data/repository/UserRepository.kt
package com.example.tradeup.data.repository

import com.example.tradeup.data.model.User

interface UserRepository {
    suspend fun createUserProfile(user: User): Result<Unit>
    suspend fun getUserProfile(uid: String): Result<User?>
    suspend fun updateUserProfile(user: User): Result<Unit>

}